navigation/image_database.h

```diff
@@ -379,11 +412,82 @@ class ImageDatabase
         return GridRecorder(*this, xrange, yrange, zrange, heading, imageFormat);
     }
 
+    //! Start recording a route
     RouteRecorder getRouteRecorder(const std::string &imageFormat = "png")
     {
         return RouteRecorder(*this, imageFormat);
     }
 
+    //! Get the resolution of saved images
+    cv::Size getResolution() const
+    {
+        BOB_ASSERT(hasMetadata());
+        return m_Resolution;
+    }
+
+    //! Check if this database has any saved metadata (yet)
+    bool hasMetadata() const { return static_cast<bool>(m_MetadataYAML); }
+
+    /**!
+     *  \brief Unwrap all the panoramic images in this database into a new
+     *         folder, creating a new database.
+     */
+    void unwrap(const filesystem::path &destination, const cv::Size &unwrapRes)
+    {
+        // Check that the database doesn't already exist
+        BOB_ASSERT(!(destination / EntriesFilename).exists());
+
+        // Create object for unwrapping images
+        std::string camName;
+        getMetadata()["camera"]["name"] >> camName;
+        ImgProc::OpenCVUnwrap360 unwrapper(getResolution(), unwrapRes, camName);
+        filesystem::create_directory(destination);
+
+        // Copy entries (CSV) file
+        copyfile(m_Path / EntriesFilename, destination / EntriesFilename);
+
+        // Create new metadata (YAML) file
+        {
+            /*
+             * There is no way to edit a persistence file in OpenCV, so we have
+             * to do it ourselves in this ugly way.
+             *
+             * First, copy all fields, except for changing needsUnwrapping to false.
+             */
+            std::string line;
+            std::ofstream ofs((destination / MetadataFilename).str());
+            for (std::ifstream ifs((m_Path / MetadataFilename).str()); std::getline(ifs, line); ) {
+                if (line.find("  needsUnwrapping:") == 0) {
```

Comment:
Might it be better to do:
```c++
const auto needsUnwrapping = line.find("needsUnwrapping:");
if(needsUnwrapping != std::string::npos) {
    ofs << std::string(needsUnwrapping, ' ') << "needsUnwrapping: 0" << std::endl;
}
```
Replying on your yaml file indentation not changing seems bad.

Reply:
Yeah that's true